api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -101,46 +103,53 @@ public List<Condition> getConditionsByEncounter(Encounter encounter) throws APIE
 	 */
 	@Override
 	public Condition saveCondition(Condition condition) throws APIException {
-		Condition existingCondition = Context.getConditionService().getConditionByUuid(condition.getUuid());
+		
 		// If there is no existing condition, then we are creating a condition
-		if (existingCondition == null) {
+		Integer existingConditionId = condition.getConditionId();
```

Comment:
Can you explain this a bit more @ibacher ?  I believe the code I have below is accomplishing the same thing, but more reliably.

Reply:
[This bit of code](https://github.com/openmrs/openmrs-module-emrapi/blob/79a5af9dac23f651b2073b413b871dd4444cc773/web-2.2/src/main/java/org/openmrs/module/emrapi/web/controller/ConditionController.java#L257). Basically, if a page POSTs to the `/rest/emrapi/condition` endpoint, it will create a "new" condition without setting the `id` on that condition, even if the condition was already an existing one; the only unique identifier it uses is the UUID.

Reply:
I see.  Yeah that'll need to get fixed.  That's an odd implementation, though consistent with the way the unit tests that I changed were operating.  Basically, you should need to fetch an existing object from the DB and edit it to update it. Creating a new object and giving it the same uuid of an existing object to reflect the desire to do an update is not a pattern I've ever seen nor would I endorse.

Reply:
I haven't dug into the reference above, but, yes, if an OpenMRS object is incoming via REST (and therefore won't have an id), is it the REST processing code that is expected to handle whether there's an existing object with the same uuid, or is this API supposed to handle it?  I think the answer is that the REST processing code generally should, and so the fix (if any is needed?) should be made in EMR-API?

Reply:
Not how I would've written it either! I was trying to keep the EMRAPI working the same way so that my two Condition PRs didn't directly depend on each other.

Reply:
OK, that is fixed and merged here https://github.com/openmrs/openmrs-module-emrapi/commit/09ddcb6132d547247d8eb0de15361ca4e8eb4350.  Thanks @ibacher  for catching and recognizing that, and for the super quick reviews.